pkg/archive/paths.go

```diff
@@ -104,4 +142,20 @@ func WalkTarGzFiles(ctx context.Context, path string, walkFn FileWalkFn) (err er
 		}
 	}
 	return nil
+
+}
+
+func CheckArchiveType(filename string) FormatType {
```

Comment:
If we return `(FormatType, error)` from here and we can provide a little more info in the error messages to aid with debugging if we hit the error cases. Or if there's no need for extra info then we can return `(FormatType, bool)` which provides a little clearer usage mechanism, `format, ok := CheckArchiveType(name)` where we can use the value of `ok` in an if statement.